lib/libc/minimal/source/stdlib/strtol.c

```diff
@@ -119,7 +119,7 @@ long strtol(const char *nptr, char **endptr, register int base)
 	if (any < 0) {
 		acc = neg ? LONG_MIN : LONG_MAX;
 		errno = ERANGE;
-	} else if (neg) {
+	} else if (neg != 0) {
```

Comment:
What about the rest in this function?

```
L96: cutoff = neg ? -(unsigned long)LONG_MIN : LONG_MAX;
L120: acc = neg ? LONG_MIN : LONG_MAX;
```

Reply:
The lines you mention do violate a different MISRA C:2012 guideline. They will be amended in a future commit and PR.

Reply:
You're missing the point -- neg should be a bool. You keep digging deeper and deeper with these casts when a type change of the variable seems like the proper approach.

Reply:
Will change neg to be a bool as soon as the branch about MISRA C:2012 Rule 10.2 has been merged.